### PR TITLE
[Skia.GTK]: Improve system dark theme detection on Linux

### LINKED ISSUE #7212: [Skia.GTK] Applications do not respond to Dark mode themes
### Current behavior

Skia.Gtk applications do not blend in when a dark theme is turned on.
image

### Expected behavior

Skia.Gtk apps should blend in.

### How to reproduce it (as minimally and precisely as possible)

Use the default Uno template application.

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

3.10.7

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2019

### IDE version

16.11

### Relevant plugins

_No response_

### Anything else we need to know?

Uno applications are currently relying on `Gtk.Settings.Default.ApplicationPreferDarkTheme`, but in practice (at least on Ubuntu), this property is never set. 

This should be used in combination with other properties:
- In convention, GTK `Dark` themes has a `-dark` suffix at its theme name.
- There are also some [hacks](https://lzone.de/blog/Detecting-a-Dark-Theme-in-GTK) using colors which might work.

### PR DESCRIPTION
**Pull request by manfromarce**
GitHub Issue (If applicable): closes #7212 

## PR Type

What kind of change does this PR introduce?

Bugfix / enhancement

## What is the current behavior?

Currently the system dark theme detection is based on the PreferDarkTheme property which is rarely used today in most distros.

## What is the new behavior?

This PR combines the PreferDarkTheme property with the theme name.
Most GTK dark themes end with -dark, or -dark- and something else in the case of Linux Mint themes. 
It is not 100% reliable but should be more accurare than the current detection.

### <samp>🤖 Generated by Copilot at 2e021be</samp>

This pull request enhances the GTK theme integration for Uno by adding a dynamic theme handler and a dark mode detection. This allows Uno apps to adapt to the GTK system theme and appearance settings.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs [hunk 1]
```diff
@@ -56,7 +58,7 @@ public SystemTheme GetSystemTheme()
 			}
 			else
 			{
-				return Settings.Default.ApplicationPreferDarkTheme ? SystemTheme.Dark : SystemTheme.Light;
+				return (Settings.Default.ApplicationPreferDarkTheme || Settings.Default.ThemeName.ToLowerInvariant().Contains("-dark")) ? SystemTheme.Dark : SystemTheme.Light;
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:61]: Consider using `Contains` with `StringComparison.OrdinalIgnoreCase`

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs [hunk 2]
```diff
@@ -29,6 +29,7 @@ private void ObserveSystemTheme()
 			{
 				var settings = Settings.Default;
 				settings.AddNotification(nameof(settings.ApplicationPreferDarkTheme), ApplicationPreferDarkThemeHandler);
+				settings.AddNotification(nameof(settings.ThemeName), ApplicationPreferDarkThemeHandler);
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:32]: I think there is a possibility that `Settings.Default.ThemeName` changes but it's still dark, in which case we will incorrectly raise SystemThemeChanged event?

Comment (manfromarce) [reply src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:32]: Yes, you're right. I will need to check and ignore cases when a light theme is changed to another light theme or dark theme to another dark theme

**Comment by jeromelaban** [comment]
Thanks for your contribution! 

Could you squash your commits and use [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) for your commit message?

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs [hunk 3]
```diff
@@ -11,9 +11,11 @@ namespace Uno.UI.Runtime.Skia.Gtk.Extensions.Helpers.Theming
 	internal class GtkSystemThemeHelperExtension : ISystemThemeHelperExtension, IDisposable
 	{
 		private bool _disposedValue;
+		private string previousThemeName = string.Empty;
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:14]:
```suggestion
		private string _previousThemeName;
```

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs [hunk 4]
```diff
@@ -11,9 +11,11 @@ namespace Uno.UI.Runtime.Skia.Gtk.Extensions.Helpers.Theming
...
 
 		internal GtkSystemThemeHelperExtension(object owner)
 		{
+			previousThemeName = Settings.Default.ThemeName;
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:18]:
```suggestion
			_previousThemeName = Settings.Default.ThemeName;
```

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs [hunk 5]
```diff
@@ -42,11 +45,24 @@ private void UnobserveSystemTheme()
...
 
 		private void ApplicationPreferDarkThemeHandler(object o, GLib.NotifyArgs args)
-			=> SystemThemeChanged?.Invoke(o, EventArgs.Empty);
+		{
+			var settings = Settings.Default;
+			if (args.Property == nameof(settings.ApplicationPreferDarkTheme) || IsGtkThemeDark(previousThemeName) != IsGtkThemeDark(settings.ThemeName))
+			{
+				SystemThemeChanged?.Invoke(o, EventArgs.Empty);
+			}
+			previousThemeName = settings.ThemeName;
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:59]:
```suggestion
			_previousThemeName = settings.ThemeName;
```

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs [hunk 6]
```diff
@@ -42,11 +45,24 @@ private void UnobserveSystemTheme()
...
 				settings.RemoveNotification(nameof(settings.ApplicationPreferDarkTheme), ApplicationPreferDarkThemeHandler);
+				settings.RemoveNotification(nameof(settings.ThemeName), ApplicationPreferDarkThemeHandler);
 			}
 		}
 
 		private void ApplicationPreferDarkThemeHandler(object o, GLib.NotifyArgs args)
-			=> SystemThemeChanged?.Invoke(o, EventArgs.Empty);
+		{
+			var settings = Settings.Default;
+			if (args.Property == nameof(settings.ApplicationPreferDarkTheme) || IsGtkThemeDark(previousThemeName) != IsGtkThemeDark(settings.ThemeName))
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Theming/GtkSystemThemeHelperExtension.cs:55]:
```suggestion
			if (args.Property == nameof(settings.ApplicationPreferDarkTheme) || IsGtkThemeDark(_previousThemeName) != IsGtkThemeDark(settings.ThemeName))
```

**Review by Youssef1313 — APPROVED** [review]

**Comment by manfromarce** [comment]
> Could you squash your commits and use [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) for your commit message?

Ok, I will do

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**